### PR TITLE
Allow to get only protected on non protected branch with github_repository_branches

### PR DESCRIPTION
**Pull request by bpaquet**
Remove leftover dead code and doc after #1117
Reintroduce the feature from #1162 with integration tests

### CODE REVIEW

**Comment by bpaquet** [comment]
@kfcampbell The feature I added in https://github.com/integrations/terraform-provider-github/pull/1162 was lost :(

File: github/data_source_github_repository_branches.go [hunk 1]
```diff
@@ -16,6 +16,16 @@ func dataSourceGithubRepositoryBranches() *schema.Resource {
 				Type:     schema.TypeString,
 				Required: true,
 			},
+			"only_protected_branches": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
```
Comment (kfcampbell) [inline github/data_source_github_repository_branches.go:22]: What do you think about adding a `ConflictsWith` for `only_protected_branches` and `only_non_protected_branches`? Currently, if both are set to true, only protected branches will be retrieved, which might be a little confusing.

File: github/data_source_github_repository_branches_test.go [hunk 2]
```diff
@@ -54,4 +54,83 @@ func TestAccGithubRepositoryBranchesDataSource(t *testing.T) {
...
+				},
+			})
+		}
+
+		t.Run("with an anonymous account", func(t *testing.T) {
+			t.Skip("anonymous account not supported for this operation")
+		})
+
+		t.Run("with an individual account", func(t *testing.T) {
+			testCase(t, individual)
```
Comment (kfcampbell) [inline github/data_source_github_repository_branches_test.go:129]: This test is failing for me for an individual account but passing for an organization account. Is it succeeding for you when running locally?

**Review by kfcampbell — COMMENTED** [review]
Thanks for catching this and following up! I had a couple quick questions, and I'm hoping we can get this into a release soon.

File: github/data_source_github_repository_branches.go (hunk 1 above)
Comment (bpaquet) [reply github/data_source_github_repository_branches.go]: Done.

File: github/data_source_github_repository_branches_test.go (hunk 2 above)
Comment (bpaquet) [reply github/data_source_github_repository_branches_test.go:129]: Locally it only run with an organization account. And is ok
```
$ TF_ACC=1 go test -v   ./... -run ^TestAccGithubRepositoryBranchesDataSource
?   	github.com/integrations/terraform-provider-github/v5	[no test files]
=== RUN   TestAccGithubRepositoryBranchesDataSource
=== RUN   TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository
=== RUN   TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository/with_an_anonymous_account
    data_source_github_repository_branches_test.go:46: anonymous account not supported for this operation
=== RUN   TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository/with_an_individual_account
    provider_utils.go:56: GITHUB_TOKEN and GITHUB_OWNER environment variables should be set
    provider_utils.go:66: Skipping TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository/with_an_individual_account which requires individual mode
=== RUN   TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository/with_an_organization_account
=== RUN   TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository_with_filtering
=== RUN   TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository_with_filtering/with_an_anonymous_account
    data_source_github_repository_branches_test.go:125: anonymous account not supported for this operation
=== RUN   TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository_with_filtering/with_an_individual_account
    provider_utils.go:56: GITHUB_TOKEN and GITHUB_OWNER environment variables should be set
    provider_utils.go:66: Skipping TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository_with_filtering/with_an_individual_account which requires individual mode
=== RUN   TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository_with_filtering/with_an_organization_account
--- PASS: TestAccGithubRepositoryBranchesDataSource (28.78s)
    --- PASS: TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository (7.96s)
        --- SKIP: TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository/with_an_anonymous_account (0.00s)
        --- SKIP: TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository/with_an_individual_account (0.00s)
        --- PASS: TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository/with_an_organization_account (7.96s)
    --- PASS: TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository_with_filtering (20.82s)
        --- SKIP: TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository_with_filtering/with_an_anonymous_account (0.00s)
        --- SKIP: TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository_with_filtering/with_an_individual_account (0.00s)
        --- PASS: TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository_with_filtering/with_an_organization_account (20.82s)
PASS
ok  	github.com/integrations/terraform-provider-github/v5/github	29.454s
```

Comment (bpaquet) [reply github/data_source_github_repository_branches_test.go:129]: Individal test is ok too:

```
$ TF_ACC=1 go test -v   ./... -run ^TestAccGithubRepositoryBranchesDataSource
?   	github.com/integrations/terraform-provider-github/v5	[no test files]
=== RUN   TestAccGithubRepositoryBranchesDataSource
=== RUN   TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository
=== RUN   TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository/with_an_anonymous_account
    data_source_github_repository_branches_test.go:46: anonymous account not supported for this operation
=== RUN   TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository/with_an_individual_account
=== RUN   TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository/with_an_organization_account
=== RUN   TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository_with_filtering
=== RUN   TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository_with_filtering/with_an_anonymous_account
    data_source_github_repository_branches_test.go:125: anonymous account not supported for this operation
=== RUN   TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository_with_filtering/with_an_individual_account
=== RUN   TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository_with_filtering/with_an_organization_account
--- PASS: TestAccGithubRepositoryBranchesDataSource (55.48s)
    --- PASS: TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository (15.47s)
        --- SKIP: TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository/with_an_anonymous_account (0.00s)
        --- PASS: TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository/with_an_individual_account (7.76s)
        --- PASS: TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository/with_an_organization_account (7.71s)
    --- PASS: TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository_with_filtering (40.01s)
        --- SKIP: TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository_with_filtering/with_an_anonymous_account (0.00s)
        --- PASS: TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository_with_filtering/with_an_individual_account (19.50s)
        --- PASS: TestAccGithubRepositoryBranchesDataSource/manages_branches_of_a_new_repository_with_filtering/with_an_organization_account (20.51s)
PASS
ok  	github.com/integrations/terraform-provider-github/v5/github	56.002s

Comment (kfcampbell) [reply github/data_source_github_repository_branches_test.go:129]: Weird...I'm getting errors when I run both (organization and individual) at the same time, but both tests are passing individually.

**Review by kfcampbell — APPROVED** [review]
Thanks for following up!

**Pull request merged**